var/spack/repos/builtin/packages/py-pyhdf/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyPyhdf(PythonPackage):
+    """pyhdf is a python wrapper around the NCSA HDF version 4 library.
+    The SD (Scientific Dataset), VS (Vdata) and V (Vgroup) APIs are
+    currently implemented. NetCDF files can also be read and modified."""
+
+    homepage = "https://github.com/fhs/pyhdf"
+    pypi = "pyhdf/pyhdf-0.10.4.tar.gz"
+    git = "https://github.com/fhs/pyhdf.git"
+
+    maintainers("climbfuji")
+
+    license("MIT")
+
+    version("master", branch="master")
+    version("0.10.4", sha256="ea09b2bdafc9be0f7f43d72ff122d8efbde61881f4da3a659b33be5e29215f93")
+
+    # Python versions
+    depends_on("python@3.2:", type=("build", "run"))
+
+    # Dependencies
+    depends_on("zlib-api", type=("build", "run"))
+    depends_on("hdf", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("jpeg", type=("build", "run"))
```

Comment:
The `zlib-api`, `jpeg`, and `hdf` dependencies aren't listed in the normal python setup files but I do see their use in the `setup_build_environment` method.

According to the `pyproject.toml` file for the version, the dependencies are:

```
requires = [  # PEP 508 specifications.
    "numpy",
    "setuptools",
    "wheel"
]
```

so you need to add `py-setuptools` as a `build` dependency since the `py-wheel` dependency is inherited.  (I don't see the `wheel` requirement for the `master` branch though.)

`setup.cfg` only for the version has:

```
[bdist_rpm]
...
requires = HDF-4.2r1
```
so I assume you'd want to specify the version range on the `hdf` dependency though I'm not sure since I'm not used to seeing it specified that way.

Reply:
Thanks. Package `hdf` only has versions `4.2.x` configured, so I will use depends_on hdf@4.2.

I will remove the Python dependency, since it is implicitly included and I don't recall where the @3.2: came from (and since spack uses 3.7:, that's all good). I'll add py-setuptools instead.